commandapi-platforms/commandapi-bukkit/commandapi-bukkit-plugin/src/main/java/dev/jorel/commandapi/CommandAPIMain.java

```diff
@@ -125,5 +129,32 @@ public void onLoad() {
 	@Override
 	public void onEnable() {
 		CommandAPI.onEnable();
+
+		// TODO: Example command, remove before merging
+		new CommandAPICommand("rate")
+			.withArguments(
+				new StringArgument("topic").setOptional(true),
+				new IntegerArgument("rating", 0, 10),
+				new PlayerArgument("target").setOptional(true)
+			)
+			.executes(info -> {
+				String topic = (String) info.args().get("topic");
+				if(topic == null) {
+					// Honestly not sure how to justify topic being optional here,
+					// causing there to be a required argument after an optional one,
+					// but I'm sure an example exists, right?
+					info.sender().sendMessage("You didn't give a rating");
```

Comment:
There is actually a use case for this! A help message stating the usage for this command. If a player types `/rate`, you could feasibly follow up with sending a message to the user stating how to use `/rate`